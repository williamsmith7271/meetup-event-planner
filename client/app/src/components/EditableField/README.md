## EditableField Component
A reusable component that shows some child content and switches to an input field when clicked to edit.

### Example

```js
<EditableField
  isEditing={isEditing}
  onClickToEdit={onClickToEdit}
  onEdit={onEditBio}
  value={bioInput || user.bio}
  name="bio"
>
  <Paragraph className={`${styles.isButton} ${styles.paragraph}`}>
    {user.bio ? user.bio : 'Click to add a bio.'}
  </Paragraph>
</EditableField>
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **isEditing**    | Bool   |             | Any bool value
| **onEdit**    | Func   |             | Any function value
| **value**    | String   |             | Any string value, represents the current input value
| **name**    | String   |             | A string that sets the label of the input, i.e. 'Bio'
| **onClickToEdit**    | Func   |             | Callback to call when the control is clicked to edit
| **children**    | Node   |             | The child component to show in non-editing state


### Other Information
Reusable, see the UserProfile for example usage.
