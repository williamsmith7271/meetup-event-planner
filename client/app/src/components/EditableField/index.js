import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Section from 'grommet-udacity/components/Section';
import Box from 'grommet-udacity/components/Box';
import FormField from 'grommet-udacity/components/FormField';

const EditableField = ({
  isEditing,
  onEdit,
  value,
  name,
  children,
}) => (
  <Section pad={{ vertical: 'medium' }} align="center" justify="center">
    {isEditing ?
      <Box size="medium" align="center">
        <FormField
          label={`Edit ${name.charAt(0).toUpperCase()}${name.slice(1)}`}
        >
          <textarea onChange={onEdit} value={value} id={`${name}-input`} />
        </FormField>
      </Box>
    :
      <span>
        {children}
      </span>
    }
  </Section>
);

EditableField.propTypes = {
  isEditing: PropTypes.bool.isRequired,
  onEdit: PropTypes.func.isRequired,
  value: PropTypes.string,
  name: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default cssModules(EditableField, styles);
