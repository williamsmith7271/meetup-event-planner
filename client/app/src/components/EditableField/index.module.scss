.formField {
  @media screen and (max-width: 700px) {
    max-width: 90vw;
    width: 400px;
  }
  @media screen and (max-width: 500px) {
    max-width: 90vw;
    width: 300px;
    textarea {
      max-width: 280px;
    }
  }
  @media screen and (max-width: 430px) {
    width: 270px;
    textarea {
      max-width: 250px;
    }
  }
  @media screen and (max-width: 400px) {
    width: 210px;
    textarea {
      max-width: 190px;
    }
  }
}
