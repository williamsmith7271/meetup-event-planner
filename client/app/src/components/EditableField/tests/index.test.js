import EditableField from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<EditableField />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <EditableField
        isEditing={false}
        onClickToEdit={e => e}
        placeholder="Hello World"
        onEdit={e => e}
        value={"hello world"}
        name="bio"
      >
        <div>
          Hi
        </div>
      </EditableField>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
