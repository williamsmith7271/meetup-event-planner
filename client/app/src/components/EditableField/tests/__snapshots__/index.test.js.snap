exports[`<EditableField /> should render with default props 1`] = `
<Section
  align="center"
  justify="center"
  pad={
    Object {
      "vertical": "medium"
    }
  }>
  <span />
</Section>
`;
